src/SqlAsyncCollector.cs

```diff
@@ -237,22 +252,41 @@ private async Task UpsertRowsAsync(IEnumerable<T> rows, SqlAttribute attribute,
         /// to upsert to and returns the extra property names in a List.
         /// </summary>
         /// <param name="columns"> The columns of the table to upsert to </param>
+        /// <param name="row"> First row to get the column names when T is JObject </param>
         /// <returns>List of property names that don't exist in the table</returns>
-        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns)
+        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns, T row)
         {
+            if (typeof(T) == typeof(JObject))
+            {
+                var jsonObj = JObject.Parse(row.ToString());
```

Comment:
Again you're doing this "turn JObject into a string and then parse it back into a JObject" - which seems unnecessary in this case since we already know it's a JObject

Reply:
again here the compiler doesn't seem to recognize the row as JObject and am not able to call the ToObject<> method. So I had to the the extra step, or else I probably should do row as JObject.